src/script/auth/util/oauthUtil.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {OAuthBody} from '@wireapp/api-client/lib/oauth/OAuthBody';
+
+import {Scope} from '../page/OAuthPermissions';
+
+/**
+ *  Takes the oauth parameters and returns the oauth object expected by the API.
+ * @param location window.location
+ * @returns
+ */
+export const oAuthParams = (location: Location) =>
+  decodeURIComponent(location.search.slice(1))
+    .split('&')
```

Comment:
can't we use URLSearchParam instead of doing the parsing ourselves? 
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
i tried this as well but it didn't seem to work very well.

Reply:
i dont recall the specifics but ill test it again to see what the issue is.